File: MqttMyApp/app/src/main/java/uk/co/falcona/mqttmyapp/MainActivity.java
package uk.co.falcona.mqttmyapp;

import android.os.Bundle;

import androidx.appcompat.app.AppCompatActivity;
import androidx.appcompat.widget.Toolbar;
import androidx.navigation.NavController;
import androidx.navigation.fragment.NavHostFragment;
import androidx.navigation.ui.NavigationUI;

public class MainActivity extends AppCompatActivity {

    private NavController navController;
    private Toolbar toolbar;
    private NavHostFragment hostFragment;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        toolbar=findViewById(R.id.tool_bar);
        setSupportActionBar(toolbar);
        hostFragment= (NavHostFragment) getSupportFragmentManager().findFragmentById(R.id.fragmentContainerView);
        navController=hostFragment.getNavController();
//        NavigationUI.setupActionBarWithNavController(this,navController);
        NavigationUI.setupWithNavController(toolbar,navController);
    }

//    @Override
//    public boolean onSupportNavigateUp() {
//        return navController.navigateUp()
//                || super.onSupportNavigateUp();
//    }
}

File: ThemingLibrary/app/src/main/java/telegram/free/theminglibrary/MainActivity.kt
package telegram.free.theminglibrary

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import androidx.appcompat.widget.Toolbar

private lateinit var toolbar:Toolbar
class MainActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        toolbar=findViewById(R.id.toolbar)
        setSupportActionBar(toolbar)
    }
}